qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -334,6 +334,82 @@ def test_rmdir_p_dir_with_file(self):
         self.mount_a.stat(self.dir_name)
 
 
+class TestLn(TestCephFSShell):
+    dir1 = 'test_dir1'
+    dir2 = 'test_dir2'
+    dump_id = 11
+
+    def test_soft_link_without_target(self):
+        s = 'somedata'
+        dump_file = 'dump' + str(self.dump_id)
+        test_script_data = f'mkdir -p {self.dir1}/{self.dir2}' \
+                           f'\ncd {self.dir1}\nput - {dump_file}' \
+                           f'\ncd {self.dir2}\nln -s ../{dump_file}' \
+                           f'\ncat {dump_file}'
```

Comment:
Execute each command individually instead in a script.

When script fails, there's not much info on what exactly failed and why::

    ======================================================================
    ERROR: test_soft_link_without_target (tasks.cephfs.test_cephfs_shell.TestLn)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/rishabh/repos/ceph/test2/qa/tasks/cephfs/test_cephfs_shell.py", line 350, in test_soft_link_without_target
        o = self.get_cephfs_shell_script_output(test_script_data, stdin=s)
      File "/home/rishabh/repos/ceph/test2/qa/tasks/cephfs/test_cephfs_shell.py", line 160, in get_cephfs_shell_script_output
        return ensure_str(self.run_cephfs_shell_script(
      File "/home/rishabh/repos/ceph/test2/qa/tasks/cephfs/test_cephfs_shell.py", line 152, in run_cephfs_shell_script
        return mount_x.client_remote.run(args=args, stdout=stdout,
      File "../qa/tasks/vstart_runner.py", line 415, in run
        return self._do_run(**kwargs)
      File "../qa/tasks/vstart_runner.py", line 454, in _do_run
        proc.wait()
      File "../qa/tasks/vstart_runner.py", line 256, in wait
        raise CommandFailedError(self.args, self.exitstatus)
    teuthology.exceptions.CommandFailedError: Command failed with status 5: 'cephfs-shell -b /tmp/test-cephfsnqyxdc3_'

Reply:
Also, next time if you want to use script, you can use dedent(). This way you make less effort to get same stuff done.

Reply:
https://github.com/ceph/ceph/blob/master/qa/tasks/cephfs/mount.py#L820-L829